### ISSUE TITLE
[🔷 Feature request ]: Derivative of Softmax

### ISSUE THREAD

**Issue by matiasvlevi** [matiasvlevi/Dann#27]
### Feature
Softmax activation function.

### Type
- [x] Dann
- [ ] Matrix
- [ ] Layer
- [x] Activation functions
- [ ] Loss functions
- [ ] Pool functions
- [ ] Datasets
- [ ] Documentation
- [ ] tests & examples
- [ ] Other 

### Description
Here is the softmax function I wrote not so long ago:
```js
/**
* Softmax function
* @method softmax
* @param z An array of numbers (vector)
* @return An array of numbers (vector)
**/
function softmax(z) {
  let ans = [];
  for (let i = 0; i < z.length; i++) {
    let top = Math.exp(z[i]);
    let sum = 0;
    for (let j = 0; j < z.length; j++) {
      sum += Math.exp(z[j]);
    }
    ans.push(top / sum);
  }
  return ans;
}
```
This function is not implemented in the repository yet.

For this function to work in a Neural Network, we would need to write the derivative of this function. This might be a difficult task since this function takes in & outputs vectors.

These two functions would need to be implemented in `src/core/functions/actfuncs.js`.

For this function to work with a Dann model, we would need to change how to activations are handled since it expects a vector instead of a number value. I could work on that once the derivative is implemented.

**Comment by hrs2203** [comment]
Hi, I would like to complete this functionality. Just one clarification, this function is meant to operator on vectors not matrices.
